wordpress-mobile/WordPress-Android
# Feature/back forward navigation

This PR adds Back and Forward navigation to Site preview and opens external links in the browser.

During my tests, Back and Forward navigation worked pretty well, but it's worth mentioning that it works on "trying it's best" basis since we don't have resources to make our own web browser, and `WebView` gives us limited tools. There are some websites that behave in a way that adds a request or sneaky redirect that we can't handle to the history, but looks like this does not compromise user experience significantly, and is a huge improvement over current behavior (Back navigation with device Back button).

PS: Navbar behavior specific to other activities that extend `WPWebViewActivity` and different use cases will be addressed in subsequent PR's.

[![Image from Gyazo](https://i.gyazo.com/bedbc6d4a4d1c26656f93566d32be979.gif)](https://gyazo.com/bedbc6d4a4d1c26656f93566d32be979)

To test:
- Open Site preview for a couple of different sites.
- Notice that clicking on links that lead to external sites opens them in an external browser (or prompts to).
- Notice that back forward navigation works mostly as expected.
- Notice that navigation back with the device back button also works.

Update release notes:
